Node 10 nan api changes and zstd 1.3.4

Stieneee opened the pull request:
Thanks for the package.

Wanted an update to date implementation of zstd running in node 10. Some nan api definitions had deprecated resulting in failed compilation.

With some stress testing I was getting some seg faults. Stack traces lead me into nan. After moving the output buffer and removing the sync worker->Destroy() call everything is behaving correctly.  I confirmed that the worker is still being destroyed properly after a sync call.

Some of the tests were missing some done calls.

zwb-ict commented:
@FGRibreau Thank you for your PR, and nice to meet you!

FGRibreau commented:
Nice work! I think I'm soon going to benchmark this lib for writing/reading data into Redis :)

Stieneee committed:
bump zstd version to 1.3.4
change to nan realted calls to support node 10
fixed incorrect tests
moved out buffers from worker
removed destroy call as was causing seg fault

Stieneee committed:
update travis to test currently maintained node envs
increase mocha timeout for slow ci performance

ethercflow reviewed (approved):
LGTM
